src/ProviderPR/SFA.DAS.ProviderPR/Application/Queries/GetEASUserByEmail/GetEasUserByEmailHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using MediatR;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests.EmployerAccounts;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses.EmployerAccounts;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.ProviderPR.Application.Queries.GetEasUserByEmail;
+public class GetEasUserByEmailQueryHandler(IAccountsApiClient<AccountsConfiguration> apiClient) : IRequestHandler<GetEasUserByEmailQuery, GetEasUserByEmailQueryResult>
+{
+
+    public async Task<GetEasUserByEmailQueryResult> Handle(GetEasUserByEmailQuery request, CancellationToken cancellationToken)
+    {
+        var result = await apiClient.Get<GetUserByEmailResponse>(new GetUserByEmailRequest(request.Email));
+
+        var userRef = result?.Ref;
+
+        if (userRef == null || userRef == Guid.Empty)
+        {
+            return new GetEasUserByEmailQueryResult { HasUserAccount = false };
+        }
+
+        var accounts = await apiClient.GetAll<GetUserAccountsResponse>(new GetUserAccountsRequest(userRef.ToString()));
+
+        if (accounts.Count() != 1)
```

Comment:
What happens if no accounts are associated with the user? We will return HasUserAccount = true. Flick the check to be `>1`. Also check for no accounts before and build response accordingly.

Reply:
am talking to Corey and yourself about this

Reply:
changed to suit.  Also checked, Corey says there will always be one or more records